Simulator/Simulator/Experiments/src/experiments/synfire_chaining.c

```diff
@@ -2,292 +2,52 @@
-	Layer* layer3_inh = layer_create_fully_connected(w_inh, n_class, "layer3_inh");
-	Layer* layer4_inh = layer_create_fully_connected(w_inh, n_class, "layer4_inh");
-	Layer* layer5_inh = layer_create_fully_connected(w_inh, n_class, "layer5_inh");
-	Layer* layer6_inh = layer_create_fully_connected(w_inh, n_class, "layer6_inh");
-	Layer* layer7_inh = layer_create_fully_connected(w_inh, n_class, "layer7_inh");
-	Layer* layer8_inh = layer_create_fully_connected(w_inh, n_class, "layer8_inh");
-	Layer* layer9_inh = layer_create_fully_connected(w_inh, n_class, "layer9_inh");
-
-	// link the neurons
-	float connectivity = 1.0f;
-	layer_add_input_layer(layer2, layer1, excitatory_s_class, connectivity);
-	layer_add_input_layer(layer2, layer1_inh, inhibitory_s_class, connectivity);
-	layer_add_input_layer(layer2_inh, layer1, excitatory_s_class, connectivity);
-	layer_add_input_layer(layer2_inh, layer1_inh, inhibitory_s_class, connectivity);
-
-	layer_add_input_layer(layer3, layer2, excitatory_s_class, connectivity);
-	layer_add_input_layer(layer3, layer2_inh, inhibitory_s_class, connectivity);
-	layer_add_input_layer(layer3_inh, layer2, excitatory_s_class, connectivity);
-	layer_add_input_layer(layer3_inh, layer2_inh, inhibitory_s_class, connectivity);
-
-	layer_add_input_layer(layer4, layer3, excitatory_s_class, connectivity);
-	layer_add_input_layer(layer4, layer3_inh, inhibitory_s_class, connectivity);
-	layer_add_input_layer(layer4_inh, layer3, excitatory_s_class, connectivity);
-	layer_add_input_layer(layer4_inh, layer3_inh, inhibitory_s_class, connectivity);
-
-	layer_add_input_layer(layer5, layer4, excitatory_s_class, connectivity);
-	layer_add_input_layer(layer5, layer4_inh, inhibitory_s_class, connectivity);
-	layer_add_input_layer(layer5_inh, layer4, excitatory_s_class, connectivity);
-	layer_add_input_layer(layer5_inh, layer4_inh, inhibitory_s_class, connectivity);
-
-	layer_add_input_layer(layer6, layer5, excitatory_s_class, connectivity);
-	layer_add_input_layer(layer6, layer5_inh, inhibitory_s_class, connectivity);
-	layer_add_input_layer(layer6_inh, layer5, excitatory_s_class, connectivity);
-	layer_add_input_layer(layer6_inh, layer5_inh, inhibitory_s_class, connectivity);
-
-	layer_add_input_layer(layer7, layer6, excitatory_s_class, connectivity);
-	layer_add_input_layer(layer7, layer6_inh, inhibitory_s_class, connectivity);
-	layer_add_input_layer(layer7_inh, layer6, excitatory_s_class, connectivity);
-	layer_add_input_layer(layer7_inh, layer6_inh, inhibitory_s_class, connectivity);
-
-	layer_add_input_layer(layer8, layer7, excitatory_s_class, connectivity);
-	layer_add_input_layer(layer8, layer7_inh, inhibitory_s_class, connectivity);
-	layer_add_input_layer(layer8_inh, layer7, excitatory_s_class, connectivity);
-	layer_add_input_layer(layer8_inh, layer7_inh, inhibitory_s_class, connectivity);
+	Network* net = NULL;
+	DataGenerator* data_gen = NULL;
+	Callback* network_dump_cb = NULL;
```

Comment:
If everything is ok you can remove this one, because it just takes a lot of time

Reply:
Removed the callback